File: task03/Haskell/main.hs
isPrime :: Int -> Bool
isPrime num
    | num <= 1 = False
    | num == 2 = True
    | even num = False
    | otherwise = all (\x -> num `mod` x /= 0) [3,5..isqrt num]
    where isqrt = floor . sqrt . fromIntegral

main :: IO ()
main = do
    putStrLn "Enter a number: "
    input <- getLine
    let n = read input :: Int
    putStrLn $ "Prime numbers up to " ++ show n ++ " are:"
    let primes = filter isPrime [2..n]
    putStrLn $ unwords (map show primes)

File: task03/c++/main.cpp
#include <iostream>
#include <cmath>

bool isPrime(int num) {
    if (num <= 1) {
        return false;
    }
    if (num == 2) {
        return true;
    }
    if (num % 2 == 0) {
        return false;
    }
    for (int i = 3; i <= sqrt(num); i += 2) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int n;
    std::cout << "Enter a number: ";
    std::cin >> n;

    std::cout << "Prime numbers up to " << n << " are:\n";
    for (int i = 2; i <= n; i++) {
        if (isPrime(i)) {
            std::cout << i << " ";
        }
    }
    std::cout << "\n";

    return 0;
}

